sw/device/boot_rom/rom_link.ld

```diff
@@ -2,106 +2,132 @@
-    ram (w)      : ORIGIN = 0x10000000, LENGTH = 0x10000
-    flash (rx)   : ORIGIN = 0x20000000, LENGTH = 0x100000
+MEMORY {
+  rom(rx)   : ORIGIN = 0x00008000, LENGTH = 0x2000
+  ram(w)    : ORIGIN = 0x10000000, LENGTH = 0x10000
+  flash(rx) : ORIGIN = 0x20000000, LENGTH = 0x100000
 }
 
-/* Memory Allocation */
+/**
+ * The boot address, which indicates the location of the initial interrupt
+ * vector.
+ */
+_boot_address = ORIGIN(rom);
+
 _heap_size = 0xe000;
 _stack_size = 0x2000;
 _stack_start = ORIGIN(ram) + _heap_size + _stack_size;
 _flash_start = ORIGIN(flash);
+
 _chip_info_size = 128;
 _chip_info_start = ORIGIN(rom) + LENGTH(rom) - _chip_info_size;
 
-/* We have to align each sector to word boundaries as our current s19->slm
- * conversion scripts are not able to handle non-word aligned sections. */
-
-SECTIONS
-{
-    .vectors :
-    {
-        . = ALIGN(4);
-        KEEP(*(.vectors))
-    } > rom
-
-    .text : {
-        . = ALIGN(4);
-        _stext = .;
-        *(.text)
-        *(.text.unlikely)
-        _etext  =  .;
-        __CTOR_LIST__ = .;
-        LONG((__CTOR_END__ - __CTOR_LIST__) / 4 - 2)
-        *(.ctors)
-        LONG(0)
-        __CTOR_END__ = .;
-        __DTOR_LIST__ = .;
-        LONG((__DTOR_END__ - __DTOR_LIST__) / 4 - 2)
-        *(.dtors)
-        LONG(0)
-        __DTOR_END__ = .;
-        *(.lit)
-        *(.shdata)
-        _endtext = .;
-    }  > rom
-
-    .rodata : {
-        . = ALIGN(4);
-        *(.rodata);
-        *(.rodata.*)
-    } > rom
-
-    .shbss :
-    {
-        . = ALIGN(4);
-        *(.shbss)
-    } > rom
-
-    .data : {
-        . = ALIGN(4);
-        sdata  =  .;
-        _sdata  =  .;
-        *(.data);
-        *(.data.*)
-        edata  =  .;
-        _edata  =  .;
-    } > rom
-
-    .chip_info _chip_info_start : {
-        . = ALIGN(4);
-        *(.chip_info)
-    } > rom
-
-    .bss :
-    {
-        . = ALIGN(4);
-        _bss_start = .;
-        *(.bss)
-        *(.bss.*)
-        *(.sbss)
-        *(.sbss.*)
-        *(COMMON)
-        _bss_end = .;
-    } > ram
-
-    .stab  0 (NOLOAD) :
-    {
-        [ .stab ]
-    }
-
-    .stabstr  0 (NOLOAD) :
-    {
-        [ .stabstr ]
-    }
+/**
+ * NOTE: We have to align each section to word boundaries as our current
+ * s19->slm conversion scripts are not able to handle non-word aligned sections.
+ */
+SECTIONS {
+  /**
+   * Ibex interrupt vector. See irq_vector.S for more information. 
+   *
+   * This has to be set up at the boot address, so that execution jumps to the
+   * reset handler correctly.
+   */
+  .vectors _boot_address : ALIGN(4) {
+    KEEP(*(.vectors))
+    *(.vectors)
+  } > rom
+
+  /**
+   * C runtime (CRT) section, containing program initialization code.
+   */
+  .crt : ALIGN(4) {
+    KEEP(*(.crt))
+    *(.crt)
+  } > rom
+
+  /**
+   * Standard text section, containing program code.
+   */
+  .text : ALIGN(4) {
+    *(.text)
+    *(.text.unlikely)
```

Comment:
Let's include `.text.hot` in this section (It might be created by the same parts of GCC that create `.text.unlikely`).